src/main/java/Duke.java

```diff
@@ -1,10 +1,88 @@
+import java.util.*;
 public class Duke {
+    private Scanner sc;
+
+    public Duke(Scanner sc) {
+        this.sc = sc;
+    }
+    public void start(Tasklist list) throws DukeException {
+        while(sc.hasNext()) {
+            String input = sc.nextLine();
+            input = input.trim();
+            String[] separateInput = input.split(" ");
+            String str = separateInput[0];
+            if(input.equals("bye")) {
```

Comment:
Perhaps you can consider defining an enum type representing the commands for this part of the code.